packages/parser/lib/Render.re

```diff
@@ -0,0 +1,191 @@
+    | NthFunction({name, payload: (selector, _)}) =>
+      ":" ++ name ++ "(" ++ render_nth_payload(selector) ++ ")"
+  and render_pseudo_selector =
+    fun
+    | Pseudoelement(v) => "::" ++ v
+    | Pseudoclass(pc) => render_pseudo_class(pc)
+  and render_compound_selector = (compound_selector: compound_selector) => {
+    let simple_selector =
+      Option.fold(
+        ~none="",
+        ~some=render_simple_selector,
+        compound_selector.type_selector,
+      );
+    let subclass_selectors =
+      List.map(render_subclass_selector, compound_selector.subclass_selectors)
+      |> String.concat("");
+    let pseudo_selectors =
+      List.map(render_pseudo_selector, compound_selector.pseudo_selectors)
+      |> String.concat("");
+    simple_selector ++ subclass_selectors ++ pseudo_selectors;
+  }
+  and render_complex_selector = complex => {
+    switch (complex) {
+    | Combinator({left, right}) =>
+      let left = render_selector(left);
+      let right = render_right_combinator(right);
+      left ++ right;
+    | Selector(selector) => render_selector(selector)
+    };
+  }
+  and render_right_combinator = right => {
+    right
+    |> List.map(((combinator, selector)) => {
+         Option.fold(~none=" ", ~some=o => " " ++ o ++ " ", combinator)
+         ++ render_selector(selector)
+       })
+    |> String.concat("");
+  }
+  and render_relative_selector = ({combinator, complex_selector}) => {
+    Option.fold(~none="", ~some=o => o ++ " ", combinator)
+    ++ render_complex_selector(complex_selector);
+  };
+
+  switch (ast) {
+  | SimpleSelector(simple) => simple |> render_simple_selector
+  | ComplexSelector(complex) => complex |> render_complex_selector
+  | CompoundSelector(compound) => compound |> render_compound_selector
+  | RelativeSelector(relative) => relative |> render_relative_selector
+  };
+}
+and render_selector_list = (ast: selector_list) => {
+  ast |> List.map(fst) |> List.map(render_selector) |> String.concat(",");
+}
+and render_component_value = (ast: component_value) => {
+  switch (ast) {
+  | Whitespace => " "
+  | Paren_block(block) => "(" ++ render_component_value_list(block) ++ ")"
+  | Bracket_block(block) => "[" ++ render_component_value_list(block) ++ "]"
+  | Percentage(string) => string ++ "%"
+  | Ident(string) => string
+  | String(string) => "\"" ++ string ++ "\""
+  | Uri(string) => "url(\"" ++ string ++ "\")"
+  | Operator(string) => string
+  | Combinator(string) => string
+  | Delim(string) => string
+  | Function(name, body) =>
+    let body = body |> fst |> render_component_value_list;
+    Printf.sprintf("%s(%s)", fst(name), body);
+  | Hash(string) => "#" ++ string
+  | Number(n) => n
+  | Unicode_range(string) => string
+  | Float_dimension((a, b)) => Printf.sprintf("%s%s", a, b)
+  | Dimension((a, b)) => Printf.sprintf("%s%s", a, b)
+  | Variable(variable) => render_variable(variable)
+  | Selector(v) => render_selector_list(v)
+  };
+};
```

Comment:
Can you remove the prefix on all the functions `render_*`? When calling it should be as simpler as Render.component_value.

Also, can you add a Render.rei where it exports only the same methods as the parser expose?

%start <stylesheet> stylesheet
%start <rule_list> declaration_list
%start <declaration> declaration
%start <rule_list> keyframes